src/main/java/org/mskcc/cbio/oncokb/web/rest/CronJobController.java

```diff
@@ -441,4 +423,37 @@ private void renewToken(Token token) {
         token.setExpiration(token.getExpiration().plusSeconds(tokenProvider.EXPIRATION_TIME_IN_SECONDS));
         tokenService.save(token);
     }
+
+    private void updateTokenStats(Instant current) {
+        log.info("Started the cronjob to update token stats");
+        List<UserTokenUsage> tokenUsages = tokenStatsService.getUserTokenUsage(current);
+
+        // Update tokens with token usage
+        tokenUsages.stream().forEach(tokenUsage -> {
+            Optional<Token> tokenOptional = tokenService.findByToken(tokenUsage.getToken().getToken());
+            if (tokenOptional.isPresent()) {
+                tokenOptional.get().setCurrentUsage(tokenOptional.get().getCurrentUsage() + tokenUsage.getCount());
+                tokenService.save(tokenOptional.get());
+            }
+        });
+        log.info("Finished the cronjob to update token stats");
+    }
+
+    private File createWrappedFile(Instant current) throws IOException {
+        File file = File.createTempFile("test-token-stats-", ".txt");
+        file.deleteOnExit();
+
+        Writer writer = new OutputStreamWriter(new FileOutputStream(file));
+        writer.write(TokenStats.csvColumns() + "\n");
+
+        int PAGE_SIZE = 1000;
+        for (int page = 0; page < tokenStatsService.findAll(current, PageRequest.of(0, PAGE_SIZE)).getTotalPages(); page++) {
+            for (TokenStats tokenStats : tokenStatsService.findAll(current, PageRequest.of(page, PAGE_SIZE))) {
+                writer.write(tokenStats.toCSV() + "\n");
+            }
+        }
+        writer.close();
```

Comment:
How many records you ran for half hour? Using the pagination is still the best option without fetching too much data into the memory. 30mins should be alright if it's processing million records.